external/js/cothority/src/byzcoin/client-transaction.ts

```diff
@@ -6,6 +6,9 @@ import Signer from "../darc/signer";
 import { EMPTY_BUFFER, registerMessage } from "../protobuf";
 import { InstanceID } from "./instance";
 
+// messages registration
```

Comment:
This feels fragile. How do we know which imports with which registration side-effects are needed in which files? Is there a way to be more systematic about this?

Reply:
It is definitely. I'm looking for a way to test this.

Reply:
So after investigating ways to force an import when there are side effects, it turns out TS guys don't want that to happen (related to static checking) so I changed the registration to declare the internal dependencies of a message so that it will correctly import every type used in the message.

I think this is better than using the previous force import because you know exactly which type is used in your message and then you simply declare it. No need to check the modules if they have side-effects.

The only confusing part is that you still need to register the messages as a side-effect because any file is an independent module and we don't want people to have to register manually when they import specific messages. I will add a paragraph in the README about that.